src/coreclr/src/jit/assertionprop.cpp

```diff
@@ -3215,6 +3215,153 @@ GenTree* Compiler::optAssertionPropGlobal_RelOp(ASSERT_VALARG_TP assertions, Gen
     GenTree* op1     = tree->AsOp()->gtOp1;
     GenTree* op2     = tree->AsOp()->gtOp2;
 
+    // First, walk up the dom tree and see if any dominating block has branched on
```

Comment:
You should update the method header to the standard header and under _Note:_ add comment saying that we also do this special case optimization.